staging/vhost-device-video/src/vhu_video.rs

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use crate::vhu_video_thread::VhostUserVideoThread;
+use crate::video_backends;
+use clap::ValueEnum;
+use log::{debug, warn};
+use std::{
+    convert,
+    io::{self, Result as IoResult},
+    path::Path,
+    sync::{Arc, Mutex, RwLock},
+};
+use thiserror::Error as ThisError;
+use vhost::vhost_user::message::{VhostUserProtocolFeatures, VhostUserVirtioFeatures};
+use vhost_user_backend::{VhostUserBackendMut, VringRwLock, VringT};
+use virtio_bindings::bindings::virtio_config::{VIRTIO_F_NOTIFY_ON_EMPTY, VIRTIO_F_VERSION_1};
+use virtio_bindings::bindings::virtio_ring::{
+    VIRTIO_RING_F_EVENT_IDX, VIRTIO_RING_F_INDIRECT_DESC,
+};
+use virtio_queue::DescriptorChain;
+use vm_memory::{ByteValued, GuestMemoryAtomic, GuestMemoryLoadGuard, GuestMemoryMmap, Le32};
+use vmm_sys_util::epoll::EventSet;
+use vmm_sys_util::eventfd::{EventFd, EFD_NONBLOCK};
+
+/// Virtio Video Feature bits
+const VIRTIO_VIDEO_F_RESOURCE_GUEST_PAGES: u16 = 0;
+/// Unsupported
+/// const VIRTIO_VIDEO_F_RESOURCE_NON_CONTIG: u16 = 1;
+/// const VIRTIO_VIDEO_F_RESOURCE_VIRTIO_OBJECT: u16 = 2;
+
+const COMMAND_Q: u16 = 0;
+const EVENT_Q: u16 = 1;
+const NUM_QUEUES: usize = 2;
+const QUEUE_SIZE: usize = 1024;
+/// Notification coming from the backend.
+/// Event range [0...num_queues] is reserved for queues and exit event.
+/// So NUM_QUEUES + 1 is used.
+pub const VIDEO_EVENT: u16 = (NUM_QUEUES + 1) as u16;
```

Comment:
By the way, `pub(crate)` is meaningless in non-library crates.

Reply:
> By the way, `pub(crate)` is meaningless in non-library crates.

Hmm, that makes sense too, since there is no lib.rs which can export stuff accidentally. So we should convert all `pub(crate)` to just `pub` ?

Reply:
It's not a hindrance just unnecessary. I am okay with just not using it in new code and leave the existing code as-is for whenever there's an opportunity for code style cleanup only changes.

By the way, in vhost-device-sound I split the crate into a library and a binary (`lib.rs` and `main.rs`) and everything except the main function is in the library, so everything is `pub`.